### PR TITLE
Rework the source load and reload.

### PR DESCRIPTION
**Pull request by knightburton**
Move the console buttons into the panel header and
extract the main logic from the connection module into the
session module.

IoT.jsCode-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

### CODE REVIEW

File: src/app/view/workspace/console.ejs [hunk 1]
```diff
@@ -4,6 +4,10 @@
     <div class="panel panel-default">
       <div class="panel-heading">
         <span>Console</span>
+        <div class="panel-heading-buttons">
+          <div class="btn btn-default btn-xs disabled" id="load-jerry-source-button" title="Load the source from the JerryScript.">Load src</div>
```
Comment (rtakacs) [inline src/app/view/workspace/console.ejs]: Load the source from JerryScript.

File: src/app/view/workspace/console.ejs [hunk 2]
```diff
@@ -4,6 +4,10 @@
...
+          <div class="btn btn-default btn-xs disabled" id="reload-jerry-source-button" title="Reload the source from the JerryScript.">Sync src</div>
```
Comment (rtakacs) [inline src/app/view/workspace/console.ejs]: Reload the source from JerryScript.

Comment (knightburton) [reply src/app/view/workspace/console.ejs]: You are right!

File: src/app/view/workspace/console.ejs (hunk 1 above)
Comment (knightburton) [reply src/app/view/workspace/console.ejs]: Indeed, thank you!

**Comment by knightburton** [comment]
@rtakacs I've updated the PR based on your comments.

File: src/app/client-connection.js [hunk 3]
```diff
@@ -295,44 +295,18 @@ function onmessage(event) {
...
-
-            if (this._surface.getPanelProperty('run.active')) {
-              this._surface.updateRunPanel(this._surface.RUN_UPDATE_TYPE.ALL, this._debuggerObj, this._session);
-            }
-
-            $(`.${groupID}`).addClass('disabled');
-            $(`.${groupID}`).unbind('click');
-            e.stopImmediatePropagation();
-          });
+          this._logger.error(`The file "${sourceName}" is missing. `, true);
```
Comment (rtakacs) [inline src/app/client-connection.js]: Please remove the extra space after the sentence.

File: src/app/view/workspace/console.ejs [hunk 4]
```diff
@@ -4,6 +4,10 @@
     <div class="panel panel-default">
       <div class="panel-heading">
         <span>Console</span>
+        <div class="panel-heading-buttons">
+          <div class="btn btn-default btn-xs disabled" id="load-jerry-source-button" title="Load the source from JerryScript.">Load src</div>
+          <div class="btn btn-default btn-xs disabled" id="reload-jerry-source-button" title="Reload the source from JerryScript.">Sync src</div>
+        </div>
```
Comment (rtakacs) [inline src/app/view/workspace/console.ejs]: What about using only one button in the console layout (e.g. a sync source button)? In this case you don't need to deal with two different buttons.
If the users click onto the sync button, the `load` or the `reload` functionality could be decided by the back-end logic (based on the `session._jerrySource` object or other helper variable).

File: src/app/session.js [hunk 5]
```diff
@@ -997,6 +1009,49 @@ export default class Session {
...
+
+        if (this._surface.getPanelProperty('run.active')) {
+          this._surface.updateRunPanel(this._surface.RUN_UPDATE_TYPE.ALL, this._debuggerObj, this._session);
+        }
+        return;
+      case SOURCE_SNYC_ACTION.RELOAD:
+        this.resetFileContent(this._jerrySource.name, this._jerrySource.source);
+        return;
+      case SOURCE_SNYC_ACTION.NOP:
+      default:
```
Comment (rtakacs) [inline src/app/session.js]: Since all cases are covered (SOURCE_SNYC_ACTION.{LOAD, RELOAD, NOP}) there is no need a default case.

Comment (knightburton) [reply src/app/session.js]: You are right, I will remove the NOP because I prefer the default case since we are using javascript.

**Comment by rtakacs** [comment]
LGTM

**Comment by yichoi** [comment]
LGTM

**Pull request merged**